Exception on application shutdown with quarkus-hibernate-search-orm-coordination-outbox-polling

### Describe the bug

When using the extension `quarkus-hibernate-search-orm-coordination-outbox-polling` and stopping the application in dev mode, exception gets thrown, caused by Hibernate Search not being able to create a connection to the database (it needs to in order to re-register agents).

### Expected behavior

No exception.

### Actual behavior

```
2023-07-05 13:57:12,576 ERROR [org.hib.sea.eng.rep.spi.RootFailureCollector] (Shutdown thread) HSEARCH000521: Hibernate Search encountered a failure during %1$s; continuing for now to list all problems, but the process will ultimately be aborted.
Context: %2$s
Failure: [Error Occurred After Shutdown]: org.hibernate.exception.GenericJDBCException: Unable to acquire JDBC Connection [This pool is closed and does not handle any more connections!] [n/a]
        at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:61)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:94)
        at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded(LogicalConnectionManagedImpl.java:116)
        at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.getPhysicalConnection(LogicalConnectionManagedImpl.java:143)
        at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.connection(StatementPreparerImpl.java:51)
        at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$1.doPrepare(StatementPreparerImpl.java:91)
        at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:177)
        at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareStatement(StatementPreparerImpl.java:76)
        at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.lambda$list$0(JdbcSelectExecutorStandardImpl.java:102)
        at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.executeQuery(DeferredResultSetAccess.java:226)
        at org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess.getResultSet(DeferredResultSetAccess.java:163)
        at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.advanceNext(JdbcValuesResultSetImpl.java:254)
        at org.hibernate.sql.results.jdbc.internal.JdbcValuesResultSetImpl.processNext(JdbcValuesResultSetImpl.java:134)
        at org.hibernate.sql.results.jdbc.internal.AbstractJdbcValues.next(AbstractJdbcValues.java:19)
        at org.hibernate.sql.results.internal.RowProcessingStateStandardImpl.next(RowProcessingStateStandardImpl.java:66)
        at org.hibernate.sql.results.spi.ListResultsConsumer.consume(ListResultsConsumer.java:178)
        at org.hibernate.sql.results.spi.ListResultsConsumer.consume(ListResultsConsumer.java:33)
        at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.doExecuteQuery(JdbcSelectExecutorStandardImpl.java:361)
        at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.executeQuery(JdbcSelectExecutorStandardImpl.java:168)
        at org.hibernate.sql.exec.internal.JdbcSelectExecutorStandardImpl.list(JdbcSelectExecutorStandardImpl.java:93)
        at org.hibernate.sql.exec.spi.JdbcSelectExecutor.list(JdbcSelectExecutor.java:31)
        at org.hibernate.loader.ast.internal.LoaderHelper.loadByArrayParameter(LoaderHelper.java:215)
        at org.hibernate.loader.ast.internal.EntityBatchLoaderArrayParam.initializeEntities(EntityBatchLoaderArrayParam.java:141)
        at org.hibernate.loader.ast.internal.EntityBatchLoaderArrayParam.load(EntityBatchLoaderArrayParam.java:98)
        at org.hibernate.loader.ast.internal.EntityBatchLoaderArrayParam.load(EntityBatchLoaderArrayParam.java:170)
        at org.hibernate.persister.entity.AbstractEntityPersister.doLoad(AbstractEntityPersister.java:3481)
        at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3471)
        at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:581)
        at org.hibernate.event.internal.DefaultLoadEventListener.loadFromCacheOrDatasource(DefaultLoadEventListener.java:567)
        at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:536)
        at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:529)
        at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:202)
        at org.hibernate.event.internal.DefaultLoadEventListener.loadWithRegularProxy(DefaultLoadEventListener.java:282)
        at org.hibernate.event.internal.DefaultLoadEventListener.proxyOrLoad(DefaultLoadEventListener.java:237)
        at org.hibernate.event.internal.DefaultLoadEventListener.doOnLoad(DefaultLoadEventListener.java:106)
        at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:78)
        at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:138)
        at org.hibernate.internal.SessionImpl.fireLoadNoChecks(SessionImpl.java:1231)
        at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1219)
        at org.hibernate.loader.internal.IdentifierLoadAccessImpl.doLoad(IdentifierLoadAccessImpl.java:194)
        at org.hibernate.loader.internal.IdentifierLoadAccessImpl.lambda$load$1(IdentifierLoadAccessImpl.java:160)
        at org.hibernate.loader.internal.IdentifierLoadAccessImpl.perform(IdentifierLoadAccessImpl.java:107)
        at org.hibernate.loader.internal.IdentifierLoadAccessImpl.load(IdentifierLoadAccessImpl.java:160)
        at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2406)
        at org.hibernate.internal.SessionImpl.find(SessionImpl.java:2372)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.cluster.impl.DefaultAgentRepository.find(DefaultAgentRepository.java:31)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.cluster.impl.AgentPersister.leaveCluster(AgentPersister.java:61)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.AbstractAgentClusterLink.leaveCluster(AbstractAgentClusterLink.java:138)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.AgentClusterLinkContextProvider.lambda$inTransaction$0(AgentClusterLinkContextProvider.java:31)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.AgentClusterLinkContextProvider.inTransaction(AgentClusterLinkContextProvider.java:42)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.AgentClusterLinkContextProvider.inTransaction(AgentClusterLinkContextProvider.java:30)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.OutboxPollingEventProcessor.leaveCluster(OutboxPollingEventProcessor.java:216)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:53)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:33)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.OutboxPollingEventProcessor.stop(OutboxPollingEventProcessor.java:211)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:53)
        at org.hibernate.search.util.common.impl.AbstractCloser.pushAll(AbstractCloser.java:92)
        at org.hibernate.search.util.common.impl.AbstractCloser.pushAll(AbstractCloser.java:73)
        at org.hibernate.search.mapper.orm.coordination.outboxpolling.impl.OutboxPollingCoordinationStrategy.stop(OutboxPollingCoordinationStrategy.java:197)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:53)
        at org.hibernate.search.mapper.orm.mapping.impl.HibernateOrmMapping.doStop(HibernateOrmMapping.java:196)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:53)
        at org.hibernate.search.util.common.impl.AbstractCloser.push(AbstractCloser.java:33)
        at org.hibernate.search.mapper.pojo.mapping.spi.AbstractPojoMappingImplementor.stop(AbstractPojoMappingImplementor.java:68)
        at org.hibernate.search.engine.common.impl.SearchIntegrationImpl.lambda$close$2(SearchIntegrationImpl.java:132)
        at org.hibernate.search.engine.common.impl.SearchIntegrationImpl.stopAllSafely(SearchIntegrationImpl.java:178)
        at org.hibernate.search.engine.common.impl.SearchIntegrationImpl.close(SearchIntegrationImpl.java:131)
        at org.hibernate.search.mapper.orm.mapping.impl.HibernateSearchContextProviderService.close(HibernateSearchContextProviderService.java:36)
        at org.hibernate.search.mapper.orm.bootstrap.impl.HibernateOrmIntegrationBooterImpl.lambda$orchestrateBootAndShutdown$5(HibernateOrmIntegrationBooterImpl.java:156)
        at java.base/java.util.concurrent.CompletableFuture.biAccept(CompletableFuture.java:1386)
        at java.base/java.util.concurrent.CompletableFuture$BiAccept.tryFire(CompletableFuture.java:1355)
        at java.base/java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1219)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
        at org.hibernate.search.mapper.orm.bootstrap.impl.HibernateSearchSessionFactoryObserver.sessionFactoryClosing(HibernateSearchSessionFactoryObserver.java:50)
        at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryClosing(SessionFactoryObserverChain.java:48)
        at org.hibernate.internal.SessionFactoryImpl.close(SessionFactoryImpl.java:869)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.close(JPAConfig.java:168)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$Destroyer.destroy(JPAConfig.java:130)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$Destroyer.destroy(JPAConfig.java:125)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.doDestroy(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.destroy(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.destroy(Unknown Source)
        at io.quarkus.arc.impl.AbstractInstanceHandle.destroyInternal(AbstractInstanceHandle.java:82)
        at io.quarkus.arc.impl.ContextInstanceHandleImpl.destroy(ContextInstanceHandleImpl.java:21)
        at io.quarkus.arc.impl.AbstractSharedContext.destroy(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:468)
        at io.quarkus.arc.Arc.shutdown(Arc.java:66)
        at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:53)
        at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:84)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
        at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
        at io.quarkus.runtime.Application.stop(Application.java:208)
        at io.quarkus.runtime.Application.stop(Application.java:155)
        at io.quarkus.runtime.ApplicationLifecycleManager$ShutdownHookThread.run(ApplicationLifecycleManager.java:436)
Caused by: java.sql.SQLException: This pool is closed and does not handle any more connections!
        at io.agroal.pool.ConnectionPool.beforeAcquire(ConnectionPool.java:209)
        at io.agroal.pool.ConnectionPool.getConnection(ConnectionPool.java:235)
        at io.agroal.pool.DataSource.getConnection(DataSource.java:86)
        at io.quarkus.hibernate.orm.runtime.customized.QuarkusConnectionProvider.getConnection(QuarkusConnectionProvider.java:23)
        at org.hibernate.internal.NonContextualJdbcConnectionAccess.obtainConnection(NonContextualJdbcConnectionAccess.java:38)
        at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded(LogicalConnectionManagedImpl.java:113)
        ... 92 more

2023-07-05 13:57:12,581 ERROR [org.hib.sea.map.orm.boo.imp.HibernateOrmIntegrationBooterImpl] (Shutdown thread) HSEARCH800035: Unable to shut down Hibernate Search: %1$s [Error Occurred After Shutdown]: org.hibernate.search.util.common.SearchException: HSEARCH000520: Hibernate Search encountered failures during shutdown. Failures:

    Hibernate ORM mapping: 
        failures: 
          - Unable to acquire JDBC Connection [This pool is closed and does not handle any more connections!] [n/a]
        at org.hibernate.search.engine.reporting.spi.RootFailureCollector.checkNoFailure(RootFailureCollector.java:53)
        at org.hibernate.search.engine.common.impl.SearchIntegrationImpl.close(SearchIntegrationImpl.java:168)
        at org.hibernate.search.mapper.orm.mapping.impl.HibernateSearchContextProviderService.close(HibernateSearchContextProviderService.java:36)
        at org.hibernate.search.mapper.orm.bootstrap.impl.HibernateOrmIntegrationBooterImpl.lambda$orchestrateBootAndShutdown$5(HibernateOrmIntegrationBooterImpl.java:156)
        at java.base/java.util.concurrent.CompletableFuture.biAccept(CompletableFuture.java:1386)
        at java.base/java.util.concurrent.CompletableFuture$BiAccept.tryFire(CompletableFuture.java:1355)
        at java.base/java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1219)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
        at org.hibernate.search.mapper.orm.bootstrap.impl.HibernateSearchSessionFactoryObserver.sessionFactoryClosing(HibernateSearchSessionFactoryObserver.java:50)
        at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryClosing(SessionFactoryObserverChain.java:48)
        at org.hibernate.internal.SessionFactoryImpl.close(SessionFactoryImpl.java:869)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.close(JPAConfig.java:168)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$Destroyer.destroy(JPAConfig.java:130)
        at io.quarkus.hibernate.orm.runtime.JPAConfig$Destroyer.destroy(JPAConfig.java:125)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.doDestroy(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.destroy(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_b44400079827d7cae558c390d8d94cadbd711498_Synthetic_Bean.destroy(Unknown Source)
        at io.quarkus.arc.impl.AbstractInstanceHandle.destroyInternal(AbstractInstanceHandle.java:82)
        at io.quarkus.arc.impl.ContextInstanceHandleImpl.destroy(ContextInstanceHandleImpl.java:21)
        at io.quarkus.arc.impl.AbstractSharedContext.destroy(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:468)
        at io.quarkus.arc.Arc.shutdown(Arc.java:66)
        at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:53)
        at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:84)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
        at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
        at io.quarkus.runtime.Application.stop(Application.java:208)
        at io.quarkus.runtime.Application.stop(Application.java:155)
```

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

We should probably shut down Agroal *after* Hibernate ORM to avoid this exception.

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmCdiProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmCdiProcessor.java
@@ -14,9 +14,7 @@ import java.util.function.Function;
 import java.util.stream.Collectors;
 
 import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.enterprise.inject.Any;
 import jakarta.enterprise.inject.Default;
-import jakarta.enterprise.inject.Instance;
 import jakarta.inject.Singleton;
 import jakarta.persistence.AttributeConverter;
 import jakarta.transaction.TransactionManager;
@@ -37,11 +35,9 @@ import org.jboss.jandex.CompositeIndex;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.FieldInfo;
 import org.jboss.jandex.MethodInfo;
-import org.jboss.jandex.ParameterizedType;
 import org.jboss.jandex.Type;
 import org.objectweb.asm.ClassVisitor;
 
-import io.agroal.api.AgroalDataSource;
 import io.quarkus.agroal.spi.JdbcDataSourceBuildItem;
 import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
 import io.quarkus.arc.deployment.AnnotationsTransformerBuildItem;
@@ -166,24 +162,7 @@ public class HibernateOrmCdiProcessor {
                 .scope(Singleton.class)
                 .unremovable()
                 .setRuntimeInit()
-                .supplier(recorder.jpaConfigSupplier())
-                .destroyer(JPAConfig.Destroyer.class);
-
-        // Add a synthetic dependency from JPAConfig to any datasource/pool,
-        // so that JPAConfig is destroyed before the datasource/pool.
-        // The alternative would be adding an application destruction observer
-        // (@Observes @BeforeDestroyed(ApplicationScoped.class)) to JPAConfig,
-        // but that would force initialization of JPAConfig upon application shutdown,
-        // which may cause cascading failures if the shutdown happened before JPAConfig was initialized.
-        if (capabilities.isPresent(Capability.HIBERNATE_REACTIVE)) {
-            configurator.addInjectionPoint(ParameterizedType.create(DotName.createSimple(Instance.class),
-                    new Type[] { ClassType.create(DotName.createSimple("io.vertx.sqlclient.Pool")) }, null),
-                    AnnotationInstance.builder(Any.class).build());
-        } else {
-            configurator.addInjectionPoint(ParameterizedType.create(DotName.createSimple(Instance.class),
-                    new Type[] { ClassType.create(DotName.createSimple(AgroalDataSource.class)) }, null),
-                    AnnotationInstance.builder(Any.class).build());
-        }
+                .supplier(recorder.jpaConfigSupplier());
 
         syntheticBeanBuildItemBuildProducer.produce(configurator.done());
     }

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -92,6 +92,7 @@ import io.quarkus.deployment.builditem.LiveReloadBuildItem;
 import io.quarkus.deployment.builditem.LogCategoryBuildItem;
 import io.quarkus.deployment.builditem.NativeImageFeatureBuildItem;
 import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
 import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
 import io.quarkus.deployment.builditem.TransformedClassesBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
@@ -678,9 +679,10 @@ public final class HibernateOrmProcessor {
     @Consume(PersistenceProviderSetUpBuildItem.class)
     @Record(RUNTIME_INIT)
     public ServiceStartBuildItem startPersistenceUnits(HibernateOrmRecorder recorder, BeanContainerBuildItem beanContainer,
-            JpaModelBuildItem jpaModel) {
+            JpaModelBuildItem jpaModel,
+            ShutdownContextBuildItem shutdownContextBuildItem) {
         if (hasEntities(jpaModel)) {
-            recorder.startAllPersistenceUnits(beanContainer.getValue());
+            recorder.startAllPersistenceUnits(beanContainer.getValue(), shutdownContextBuildItem);
         }
 
         return new ServiceStartBuildItem("Hibernate ORM");

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java
@@ -36,6 +36,7 @@ import io.quarkus.hibernate.orm.runtime.proxies.PreGeneratedProxies;
 import io.quarkus.hibernate.orm.runtime.schema.SchemaManagementIntegrator;
 import io.quarkus.hibernate.orm.runtime.tenant.DataSourceTenantConnectionResolver;
 import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.ShutdownContext;
 import io.quarkus.runtime.annotations.Recorder;
 
 /**
@@ -106,8 +107,16 @@ public class HibernateOrmRecorder {
         return () -> new JPAConfig(runtimeConfig.getValue());
     }
 
-    public void startAllPersistenceUnits(BeanContainer beanContainer) {
-        beanContainer.beanInstance(JPAConfig.class).startAll();
+    public void startAllPersistenceUnits(BeanContainer beanContainer, ShutdownContext shutdownContext) {
+        JPAConfig jpaConfig = beanContainer.beanInstance(JPAConfig.class);
+        // NOTE:
+        //  - We register the shutdown task before we start any PUs,
+        //    This way we'll be able to clean up even if one of the PUs fails to start while others already did.
+        //  - The step that starts the PUs returns the ServiceStartBuildItem, this in turn ensures that
+        //    the shutdown task that triggers the ShutdownEvent will be registered after this one,
+        //    and users will have access to the "ORM stuff" in their listeners.
+        shutdownContext.addShutdownTask(jpaConfig::shutdown);
+        jpaConfig.startAll();
     }
 
     public Function<SyntheticCreationalContext<SessionFactory>, SessionFactory> sessionFactorySupplier(

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java
@@ -10,14 +10,12 @@ import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
-import jakarta.enterprise.context.spi.CreationalContext;
 import jakarta.inject.Inject;
 import jakarta.persistence.EntityManagerFactory;
 import jakarta.persistence.Persistence;
 
 import org.jboss.logging.Logger;
 
-import io.quarkus.arc.BeanDestroyer;
 import io.quarkus.hibernate.orm.runtime.boot.QuarkusPersistenceUnitDescriptor;
 
 public class JPAConfig {
@@ -129,18 +127,22 @@ public class JPAConfig {
         return this.requestScopedSessionEnabled;
     }
 
-    public static class Destroyer implements BeanDestroyer<JPAConfig> {
-        @Override
-        public void destroy(JPAConfig instance, CreationalContext<JPAConfig> creationalContext, Map<String, Object> params) {
-            for (LazyPersistenceUnit factory : instance.persistenceUnits.values()) {
+    void shutdown() {
+        LOGGER.trace("Starting to shut down Hibernate ORM persistence units.");
+        for (LazyPersistenceUnit factory : this.persistenceUnits.values()) {
+            if (factory.isStarted()) {
                 try {
+                    LOGGER.tracef("Closing Hibernate ORM persistence unit: %s.", factory.name);
                     factory.close();
                 } catch (Exception e) {
                     LOGGER.warn("Unable to close the EntityManagerFactory: " + factory, e);
                 }
+            } else {
+                LOGGER.tracef("Skipping Hibernate ORM persistence unit, that failed to start: %s.", factory.name);
             }
-            instance.persistenceUnits.clear();
         }
+        this.persistenceUnits.clear();
+        LOGGER.trace("Finished shutting down Hibernate ORM persistence units.");
     }
 
     static final class LazyPersistenceUnit {
@@ -174,6 +176,10 @@ public class JPAConfig {
             if (emf != null) {
                 emf.close();
             }
+        }
+
+        boolean isStarted() {
+            return !closed && value != null;
         }
     }
 

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/pom.xml
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/pom.xml
@@ -33,6 +33,11 @@
         <!-- test dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-deployment</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit5-internal</artifactId>
             <scope>test</scope>
         </dependency>
@@ -49,6 +54,11 @@
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
             <scope>test</scope>
         </dependency>
     </dependencies>
@@ -77,6 +87,32 @@
                 <configuration>
                     <skip>true</skip>
                 </configuration>
+                <executions>
+                    <execution>
+                        <id>default-test</id>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <!-- These tests seem to leak metaspace memory
+                                 so we run them in a separate, short-lived JVM -->
+                            <excludedGroups>devmode</excludedGroups>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>devmode-test</id>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <!-- These tests seem to leak metaspace memory
+                                 so we run them in a separate, short-lived JVM -->
+                            <groups>devmode</groups>
+                            <!-- We could consider setting reuseForks=false if we
+                                 really need to run every single test in its own JVM -->
+                        </configuration>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeFailingOrmTest.java
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeFailingOrmTest.java
@@ -1,0 +1,82 @@
+package io.quarkus.hibernate.search.orm.outboxpolling.test.configuration.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+@Tag("devmode")
+public class HibernateSearchDevModeFailingOrmTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(HibernateSearchOutboxPollingTestResource.class,
+                            HibernateSearchOutboxPollingTestResource.Person.class,
+                            HibernateSearchOutboxPollingTestResource.OutboxPollingTestUtils.class)
+                    .addAsResource("application-dev-mode.properties", "application.properties"))
+            .setLogRecordPredicate(r -> true);
+
+    static String APPLICATION_PROPERTIES;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        APPLICATION_PROPERTIES = Files
+                .readString(
+                        Paths.get(HibernateSearchDevModeFailingOrmTest.class.getResource("/application-dev-mode.properties")
+                                .toURI()));
+    }
+
+    @Test
+    public void smoke() {
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(200);
+
+        // now add a property that will fail the datasource:
+        config.modifyResourceFile(
+                "application.properties",
+                s -> APPLICATION_PROPERTIES + """
+                        quarkus.datasource.jdbc.min-size=20
+                        """);
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(500);
+
+        // add any change to get the shutdown of a failed app completed:
+        config.modifyResourceFile("application.properties", s -> APPLICATION_PROPERTIES);
+
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(200);
+
+        // At this point we've tried starting the app 3 times: one initial, one failing, one successful live-reloads
+        // Hence we expect the following things logged:
+        // initial run:
+        //  - profile activated (after a successful startup)
+        //  - ORM message after a successful shutdown caused by a following live-reload (closing a PU)
+        // first reload:
+        //  - ORM message telling us that the PU closing won't happen as the PU failed to start
+        // second reload:
+        //  - profile activated (after a successful startup)
+        //  - no ORM shutdown message, as that will happen after the test body finishes.
+        assertThat(config.getLogRecords()).satisfiesOnlyOnce(
+                r -> {
+                    assertThat(r.getMessage()).contains("Closing Hibernate ORM persistence unit");
+                    assertThat(r.getParameters()).containsExactly("<default>");
+                });
+        assertThat(config.getLogRecords()).satisfiesOnlyOnce(
+                r -> {
+                    assertThat(r.getMessage()).contains("Skipping Hibernate ORM persistence unit, that failed to start");
+                    assertThat(r.getParameters()).containsExactly("<default>");
+                });
+        assertThat(config.getLogRecords().stream()
+                .filter(r -> r.getMessage().contains("Profile%s %s activated. %s")))
+                .hasSize(2);
+    }
+}

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeFailingSearchTest.java
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeFailingSearchTest.java
@@ -1,0 +1,82 @@
+package io.quarkus.hibernate.search.orm.outboxpolling.test.configuration.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+@Tag("devmode")
+public class HibernateSearchDevModeFailingSearchTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(HibernateSearchOutboxPollingTestResource.class,
+                            HibernateSearchOutboxPollingTestResource.Person.class,
+                            HibernateSearchOutboxPollingTestResource.OutboxPollingTestUtils.class)
+                    .addAsResource("application-dev-mode.properties", "application.properties"))
+            .setLogRecordPredicate(r -> true);
+
+    static String APPLICATION_PROPERTIES;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        APPLICATION_PROPERTIES = Files
+                .readString(Paths
+                        .get(HibernateSearchDevModeFailingSearchTest.class.getResource("/application-dev-mode.properties")
+                                .toURI()));
+    }
+
+    @Test
+    public void smoke() {
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(200);
+
+        // now add a property that will fail the search, but since search is started through ORM integrator.. we are actually failing ORM startup:
+        config.modifyResourceFile(
+                "application.properties",
+                s -> APPLICATION_PROPERTIES.replace(
+                        "quarkus.hibernate-search-orm.elasticsearch.hosts=${elasticsearch.hosts:localhost:9200}",
+                        "quarkus.hibernate-search-orm.elasticsearch.hosts=not-a-localhost:9211"));
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(500);
+
+        // and any change to get the shutdown of a failed app completed:
+        config.modifyResourceFile("application.properties", s -> APPLICATION_PROPERTIES);
+
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(200);
+
+        // At this point we've tried starting the app 3 times: one initial, one failing, one successful live-reloads
+        // Hence we expect the following things logged:
+        // initial run:
+        //  - profile activated (after a successful startup)
+        //  - ORM message after a successful shutdown caused by a following live-reload (closing a PU)
+        // first reload:
+        //  - ORM message telling us that the PU closing won't happen as the PU failed to start
+        // second reload:
+        //  - profile activated (after a successful startup)
+        //  - no ORM shutdown message, as that will happen after the test body finishes.
+        assertThat(config.getLogRecords()).satisfiesOnlyOnce(
+                r -> {
+                    assertThat(r.getMessage()).contains("Closing Hibernate ORM persistence unit");
+                    assertThat(r.getParameters()).containsExactly("<default>");
+                });
+        assertThat(config.getLogRecords()).satisfiesOnlyOnce(
+                r -> {
+                    assertThat(r.getMessage()).contains("Skipping Hibernate ORM persistence unit, that failed to start");
+                    assertThat(r.getParameters()).containsExactly("<default>");
+                });
+        assertThat(config.getLogRecords().stream()
+                .filter(r -> r.getMessage().contains("Profile%s %s activated. %s")))
+                .hasSize(2);
+    }
+}

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeTest.java
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchDevModeTest.java
@@ -1,0 +1,49 @@
+package io.quarkus.hibernate.search.orm.outboxpolling.test.configuration.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+@Tag("devmode")
+public class HibernateSearchDevModeTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(HibernateSearchOutboxPollingTestResource.class)
+                    .addAsResource("application.properties", "application.properties"))
+            .setLogRecordPredicate(r -> true);
+
+    @Test
+    public void smoke() {
+        String[] schemaManagementStrategies = { "drop-and-create-and-drop", "drop-and-create" };
+
+        RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                .statusCode(200)
+                .body(is("OK"));
+
+        for (int i = 0; i < 3; i++) {
+            int current = i;
+            config.modifyResourceFile(
+                    "application.properties",
+                    s -> s.replace(
+                            "quarkus.hibernate-search-orm.schema-management.strategy="
+                                    + schemaManagementStrategies[current % 2],
+                            "quarkus.hibernate-search-orm.schema-management.strategy="
+                                    + schemaManagementStrategies[(current + 1) % 2]));
+
+            RestAssured.when().put("/test/hibernate-search-outbox-polling/check-agents-running").then()
+                    .statusCode(200)
+                    .body(is("OK"));
+        }
+
+        assertThat(config.getLogRecords()).noneSatisfy(
+                r -> assertThat(r.getMessage()).contains("Unable to shut down Hibernate Search"));
+    }
+}

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchOutboxPollingTestResource.java
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/java/io/quarkus/hibernate/search/orm/outboxpolling/test/configuration/devmode/HibernateSearchOutboxPollingTestResource.java
@@ -1,0 +1,249 @@
+package io.quarkus.hibernate.search.orm.outboxpolling.test.configuration.devmode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.Duration;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.transaction.HeuristicMixedException;
+import jakarta.transaction.HeuristicRollbackException;
+import jakarta.transaction.NotSupportedException;
+import jakarta.transaction.RollbackException;
+import jakarta.transaction.SystemException;
+import jakarta.transaction.Transactional;
+import jakarta.transaction.UserTransaction;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import org.hibernate.search.engine.backend.types.Sortable;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.outboxpolling.cluster.impl.Agent;
+import org.hibernate.search.mapper.orm.outboxpolling.cluster.impl.AgentState;
+import org.hibernate.search.mapper.orm.outboxpolling.cluster.impl.OutboxPollingAgentAdditionalMappingProducer;
+import org.hibernate.search.mapper.orm.outboxpolling.event.impl.OutboxEvent;
+import org.hibernate.search.mapper.orm.outboxpolling.event.impl.OutboxPollingOutboxEventAdditionalMappingProducer;
+import org.hibernate.search.mapper.orm.session.SearchSession;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.KeywordField;
+
+@Path("/test/hibernate-search-outbox-polling")
+public class HibernateSearchOutboxPollingTestResource {
+
+    @Inject
+    EntityManager entityManager;
+
+    @Inject
+    UserTransaction userTransaction;
+
+    @PUT
+    @Path("/check-agents-running")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String checkAgentsRunning() {
+        OutboxPollingTestUtils.awaitAgentsRunning(entityManager, userTransaction, 1);
+        return "OK";
+    }
+
+    @PUT
+    @Path("/init-data")
+    @Transactional
+    public void initData() {
+        createPerson("John Irving");
+        createPerson("David Lodge");
+        createPerson("Paul Auster");
+        createPerson("John Grisham");
+
+        // Add many other entities, so that mass indexing has something to do.
+        // DO NOT REMOVE, it's important to have many entities to fully test mass indexing.
+        for (int i = 0; i < 1000; i++) {
+            createPerson("Other entity #" + i);
+        }
+    }
+
+    @PUT
+    @Path("/await-event-processing")
+    public void awaitEventProcessing() {
+        OutboxPollingTestUtils.awaitNoMoreOutboxEvents(entityManager, userTransaction);
+    }
+
+    @GET
+    @Path("/search")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public String testSearch() {
+        SearchSession searchSession = Search.session(entityManager);
+
+        List<Person> person = searchSession.search(Person.class)
+                .where(f -> f.match().field("name").matching("john"))
+                .sort(f -> f.field("name_sort"))
+                .fetchHits(20);
+
+        assertEquals(2, person.size());
+        assertEquals("John Grisham", person.get(0).getName());
+        assertEquals("John Irving", person.get(1).getName());
+
+        assertEquals(4 + 1000, searchSession.search(Person.class)
+                .where(f -> f.matchAll())
+                .fetchTotalHitCount());
+
+        return "OK";
+    }
+
+    @PUT
+    @Path("/purge")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String testPurge() {
+        SearchSession searchSession = Search.session(entityManager);
+
+        searchSession.workspace().purge();
+
+        return "OK";
+    }
+
+    @PUT
+    @Path("/refresh")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String testRefresh() {
+        SearchSession searchSession = Search.session(entityManager);
+
+        searchSession.workspace().refresh();
+
+        return "OK";
+    }
+
+    @GET
+    @Path("/search-empty")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public String testSearchEmpty() {
+        SearchSession searchSession = Search.session(entityManager);
+
+        List<Person> person = searchSession.search(Person.class)
+                .where(f -> f.matchAll())
+                .fetchHits(20);
+
+        assertEquals(0, person.size());
+
+        return "OK";
+    }
+
+    @PUT
+    @Path("/mass-indexer")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String testMassIndexer() throws InterruptedException {
+        SearchSession searchSession = Search.session(entityManager);
+
+        searchSession.massIndexer().startAndWait();
+
+        return "OK";
+    }
+
+    private void createPerson(String name) {
+        Person entity = new Person(name);
+        entityManager.persist(entity);
+    }
+
+    @Entity
+    @Indexed
+    public static class Person {
+
+        @Id
+        @GeneratedValue
+        private Long id;
+
+        @FullTextField
+        @KeywordField(name = "name_sort", sortable = Sortable.YES)
+        private String name;
+
+        Person() {
+        }
+
+        public Person(String name) {
+            this.name = name;
+        }
+
+        public Long getId() {
+            return id;
+        }
+
+        public void setId(Long id) {
+            this.id = id;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+    }
+
+    public static class OutboxPollingTestUtils {
+
+        public static void inTransaction(UserTransaction transaction, Runnable runnable) {
+            try {
+                transaction.begin();
+                try {
+                    runnable.run();
+                    transaction.commit();
+                } catch (Throwable t) {
+                    try {
+                        transaction.rollback();
+                    } catch (Throwable t2) {
+                        t.addSuppressed(t2);
+                    }
+                    throw t;
+                }
+            } catch (SystemException | NotSupportedException | RollbackException | HeuristicMixedException
+                    | HeuristicRollbackException e) {
+                throw new IllegalStateException("Transaction exception", e);
+            }
+        }
+
+        public static void awaitAgentsRunning(EntityManager entityManager, UserTransaction userTransaction,
+                int expectedAgentCount) {
+            await("Waiting for the formation of a cluster of " + expectedAgentCount + " agents")
+                    .pollDelay(Duration.ZERO)
+                    .pollInterval(Duration.ofMillis(5))
+                    .atMost(Duration.ofSeconds(10)) // CI can be rather slow...
+                    .untilAsserted(() -> inTransaction(userTransaction, () -> {
+                        List<Agent> agents = entityManager
+                                .createQuery("select a from " + OutboxPollingAgentAdditionalMappingProducer.ENTITY_NAME
+                                        + " a order by a.id", Agent.class)
+                                .getResultList();
+                        assertThat(agents)
+                                .hasSize(expectedAgentCount)
+                                .allSatisfy(agent -> {
+                                    assertThat(agent.getState()).isEqualTo(AgentState.RUNNING);
+                                    assertThat(agent.getTotalShardCount()).isEqualTo(expectedAgentCount);
+                                });
+                    }));
+        }
+
+        public static void awaitNoMoreOutboxEvents(EntityManager entityManager, UserTransaction userTransaction) {
+            await("Waiting for all outbox events to be processed")
+                    .pollDelay(Duration.ZERO)
+                    .pollInterval(Duration.ofMillis(5))
+                    .atMost(Duration.ofSeconds(20)) // CI can be rather slow...
+                    .untilAsserted(() -> inTransaction(userTransaction, () -> {
+                        List<OutboxEvent> events = entityManager
+                                .createQuery("select e from " + OutboxPollingOutboxEventAdditionalMappingProducer.ENTITY_NAME
+                                        + " e order by e.id", OutboxEvent.class)
+                                .getResultList();
+                        assertThat(events).isEmpty();
+                    }));
+        }
+
+    }
+}

--- a/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/resources/application-dev-mode.properties
+++ b/extensions/hibernate-search-orm-outbox-polling/deployment/src/test/resources/application-dev-mode.properties
@@ -1,0 +1,16 @@
+quarkus.ssl.native = false
+
+quarkus.datasource.db-kind=h2
+quarkus.datasource.jdbc.url=jdbc:h2:mem:default;DB_CLOSE_DELAY=-1
+quarkus.datasource.jdbc.max-size=8
+
+quarkus.hibernate-orm.database.generation=drop-and-create
+
+quarkus.hibernate-search-orm.elasticsearch.version=9
+quarkus.hibernate-search-orm.elasticsearch.hosts=${elasticsearch.hosts:localhost:9200}
+quarkus.hibernate-search-orm.elasticsearch.protocol=${elasticsearch.protocol:http}
+quarkus.hibernate-search-orm.schema-management.strategy=drop-and-create-and-drop
+quarkus.hibernate-search-orm.coordination.strategy=outbox-polling
+
+quarkus.log.category."io.quarkus.hibernate.orm.runtime".min-level=TRACE
+quarkus.log.category."io.quarkus.hibernate.orm.runtime".level=TRACE

